trio/_threads.py

```diff
@@ -170,6 +236,9 @@ async def to_thread_run_sync(  # type: ignore[misc]
     # for the result – or None if this function was cancelled and we should
     # discard the result.
     task_register: list[trio.lowlevel.Task | None] = [trio.lowlevel.current_task()]
+    # Holds a reference to the raise_cancel function provided if a cancellation
+    # is attempted against this task - or None if no such delivery has happened.
+    cancel_register: list[RaiseCancelT | None] = [None]  # type: ignore[assignment]
```

Comment:
Again, I'm new at this and my local mypy just told me that I was a bad person for trying to assign this. On it's face, and in practical execution, it seems fine. I didn't try sending it to CI though, so maybe it could be a local machine issue. If you remove the ignore, does it work locally for you?

Reply:
Seems like a mypy bug to me; you can't assign `list[None]` to `list[T | None]` (because you could use the more general type to append a T, violating the type of the original) but you should be able to assign an actual literal `[None]` to the latter because it doesn't have any other references. And it worked three lines ago in an analogous situation. mypy will warn us if the type-ignore is no longer needed (we are using strict mode, right?) so fine by me to leave it for now.

Reply:
> mypy will warn us if the type-ignore is no longer needed (we are using strict mode, right?)

`strict=True` is not literally set (though I realize now it probably should be to automatically enable any future checks added by mypy), but `warn_unused_ignores` is, so yeah it will give a warning.